Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -963,8 +988,82 @@ def get_mapping_fields_command(service):
+
+    Args:
+        last_update: The time to be transformed
+
+    Returns: The corresponding time on the Splunk server
+
+    """
+    last_update_utc_datetime = dateparser.parse(last_update, settings={'TIMEZONE': 'UTC'})
+    dem_params = demisto.params()
+    splunk_timezone = dem_params.get('timezone')
+    if not splunk_timezone:
+        raise Exception('Cannot mirror incidents when timezone is not configured. Please enter the '
+                        'timezone of the Splunk server being used in the integration configuration.')
+    return date_to_timestamp(last_update_utc_datetime + timedelta(minutes=int(splunk_timezone))) / 1000
+
+
+def get_remote_data_command(service):
+    notable_updated_data = []
+    entries = []
+    updated_notable = {}
+    remote_args = GetRemoteDataArgs(demisto.args())
+    last_update_splunk_timestamp = get_last_update_in_splunk_time(remote_args.last_update)
+    notable_id = remote_args.remote_incident_id
+    search = '|`incident_review` ' \
+             '| eval last_modified_timestamp=_time ' \
+             '| where rule_id="{}" ' \
+             '| where last_modified_timestamp>{} ' \
+             '| fields - time ' \
+             '| map search=" search `notable_by_id($rule_id$)`"'.format(notable_id, last_update_splunk_timestamp)
+    demisto.debug('Performing get-remote-data command with query: {}'.format(search))
+
+    for item in results.ResultsReader(service.jobs.oneshot(search)):
+        notable_updated_data.append(parse_notable(item, to_dict=True))
+
+    if notable_updated_data:
+        updated_notable = notable_updated_data[-1]  # a notable can change several times, so taking the latest
+
+    if updated_notable.get('status') == '5' and demisto.params().get('close_incident'):
+        demisto.debug('Closing incident related to notable {}'.format(notable_id))
+        entries.append({
+            'Type': EntryType.NOTE,
+            'Contents': {
+                'dbotIncidentClose': True,
+                'closeReason': 'Notable event was closed on Splunk.'
+            },
+            'ContentsFormat': EntryFormat.JSON
+        })
+
+    demisto.info('Updated notable {}'.format(notable_id))
+    return GetRemoteDataResponse(
+        mirrored_object=updated_notable,
+        entries=entries
+    )
+
+
+def get_modified_remote_data_command(service):
+    modified_notable_ids = []
+    remote_args = GetModifiedRemoteDataArgs(demisto.args())
+    last_update_splunk_timestamp = get_last_update_in_splunk_time(remote_args.last_update)
+
+    search = '|`incident_review` ' \
+             '| eval last_modified_timestamp=_time ' \
+             '| where last_modified_timestamp>{} ' \
+             '| fields - time'.format(last_update_splunk_timestamp)
+    demisto.debug('Performing get-modified-remote-data command with query: {}'.format(search))
+    for item in results.ResultsReader(service.jobs.oneshot(search)):
+        modified_notable_ids.append(item['rule_id'])
```

Comment:
how confident are we that `rule_id` will be in `item` ? wouldnt it be better to stay on the safe side and add a condition here?

Reply:
The \`incident_review\` macro is construced from the `| inputlookup append=T incident_review_lookup` search which always returns the `rule_id`. We can be on the safe side although it's not needed.